AltStore/Operations/ResignAppOperation.swift

```diff
@@ -113,11 +113,21 @@ private extension ResignAppOperation
             
             infoDictionary[kCFBundleIdentifierKey as String] = profile.bundleIdentifier
             infoDictionary[Bundle.Info.altBundleID] = identifier
-            
+
             for (key, value) in additionalInfoDictionaryValues
             {
                 infoDictionary[key] = value
             }
+
+            if var extensionInfo = infoDictionary["NSExtension"] as? [String: Any],
+                let extensionType = extensionInfo["NSExtensionPointIdentifier"] as? String,
+                let appGroup = extensionInfo["NSExtensionFileProviderDocumentGroup"] as? String,
+                let teamId = profile.entitlements[.teamIdentifier] as? String,
+                extensionType == "com.apple.fileprovider-nonui"
+            {
+                extensionInfo["NSExtensionFileProviderDocumentGroup"] = appGroup + "." + teamId
```

Comment:
No guarantees AltStore's format for resigned app groups will remain the same in the future, so best not to hardcode the format but rather re-use the same app group from the provisioning profile. I'd be okay with just assuming the first one is correct for now, but you could also find the correct app group by checking which resigned App ID contains the original app ID (which I do plan to keep consistent, regardless of actual format).

```suggestion
                let appGroups = profile.entitlements[.appGroups] as? [String],
                let appGroup = appGroups.first,
                let teamId = profile.entitlements[.teamIdentifier] as? String,
                extensionType == "com.apple.fileprovider-nonui"
            {
                extensionInfo["NSExtensionFileProviderDocumentGroup"] = appGroup
```

Reply:
I'll try taking the shortest app group with a prefix match. For example, if NSExtensionFileProviderDocumentGroup is `group.foo.bar` and you have both `group.foo.bar.ABC123` and `group.foo.bar.baz.ABC123`, choose the shorter one.